neuro_san/internals/chat/chat_history_message_processor.py

```diff
@@ -25,10 +26,24 @@ class ChatHistoryMessageProcessor(MessageProcessor):
     in a chat history.
     """
 
-    def __init__(self):
+    def __init__(self, max_message_history: int = None):
         """
         Constructor
+
+        :param max_message_history: The maximum number of messages to preserve
+                in the message history, not including the instructions.
+                The default value of None implies there is no maximum.
+                Non-positive numbers revert to no-maximum behavior.
         """
+        self.max_message_history: int = max_message_history
```

Comment:
Not a biggie, but you can initialize the field to None, and set it if the input parameter is an instance of a Number (and have an if instead of if/else)